test/dtypes/test_nf4.py

```diff
@@ -222,7 +223,98 @@ def test_smoketest_linear_compile(self, dtype: torch.dtype):
         out3 = torch.compile(torch.nn.functional.linear, mode='max-autotune')(inp, a_nf4)
 
 
+class TestFSDPOps(TestCase):
+    @parametrize("input_size", [512 * 512, (512 * 512,), (512, 512)])
+    def test_torch_chunk_valid(self, input_size: Union[Tuple[int], int]):
+        num_chunks = 2
+        nf4_tensor = to_nf4(torch.randn(input_size))
+        chunks = list(torch.chunk(nf4_tensor, num_chunks))
+        self.assertEqual(len(chunks), num_chunks)
+        if isinstance(input_size, int):
+            expected_size0 = input_size // num_chunks
+        else:
+            expected_size0 = input_size[0] // num_chunks
+        for chunk in chunks:
+            self.assertEqual(chunk.size(0), expected_size0)
+
+    @parametrize("input_size", [511 * 512, (511 * 512,), (511, 512), (512, 512, 512)])
+    def test_torch_chunk_invalid(self, input_size: Union[Tuple[int], int]):
+        num_chunks = 2
+        with self.assertRaises(AssertionError):
```

Comment:
nit: maybe add the message to the assert message, it is not immediately apparent at least to my why these chunks are invalid

Reply:
haha. it's from NF4Tensor constructor and complains about non-dividble block_size . I can add detailed errror msg there